control-plane/api/v1alpha1/httprouteretryfilter_types.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&HTTPRouteRetryFilter{}, &HTTPRouteRetryFilterList{})
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// HTTPRouteRetryFilter is the Schema for the httprouteretryfilters API
+// +kubebuilder:printcolumn:name="Synced",type="string",JSONPath=".status.conditions[?(@.type==\"Synced\")].status",description="The sync status of the resource with Consul"
+// +kubebuilder:printcolumn:name="Last Synced",type="date",JSONPath=".status.lastSyncedTime",description="The last successful synced time of the resource with Consul"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="The age of the resource"
+type HTTPRouteRetryFilter struct {
```

Comment:
Similar to the conversation around naming with @jm96441n's PR, should this be something more generic like `RouteRetyFilter` assuming we'd use it on GRPC-based routes (if that's a thing in envoy) eventually?

Reply:
Ok, I'll update it and regen